Reverts "Tweaking stuff" vault removal, Keeps other changes.

DeveloperLoser opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Simply sets the vault slots back to what they were minus some slots.

## Why It's Good For The Game

The vault is generally disliked for how the people within power game, and in the case of the PR that removed it, clearly titled as "Tweaking Stuff" to many job slots was sited as a problem.

First power gaming, vault generally keeps to itself and if they do ruffle feathers, are easy to kill/cut off from the surface. and while they have access to some pretty mean tech, so do the BOS and Enclave, who also get better armor, and the components to easily craft weapons en-mass. Even then, if the vault runs up to the surface, murders the fucking shit out of people with their max-caps (also easy to remove TTV valves), they should be yelled at by admins for over-escalation, or told off for power gaming.

Secondly slots. The vault does have a stupid fucking amount of slots coming in at 19, that has been changed, mainly reducing the number of engineers and dwellers. Additionally, the servers population has rapidly increased, and we are now regularly near the top of the server list. If the need to decrease the number of people in vault more is felt, then that can be easily done, but it shouldn't just be completely removed, especially in an un-tagged, badly named PR.

Finally vault is a great place to learn and play around. Its a safe, secure bunker in the corner of the darkest part of the map where you can have fun conversations, play around with whatever bullshit you want, and generally learn and have fun.

Also, if this PR gets merged then I'll be more than happy to expand on the vaults content, like
- giving them station like objectives. (something to do for the round, tg vault moment)
- starting the vault in a state of decay, requiring the dwellers to repair it. (more on the teaching new players side)
- Remove TTV's from the shuttle, as well as virus crats. (if they aren't already, which I've heard mixed info on)

Since there were quite a few slots, here's how I've changed/removed some. (Because details fucking matter!)

- Overseer:                
       - 1 Before
       - 1 After
       -  Vault needs a leader (If someone ever decides to be one)
- Chief of Security:   
       - 1 Before
       -  0 After
       - The overseer works fine as a HOS.
- Doctor:                  
       - 2 Before
       - 2 After
       - One for actually treating people, the other is just a chemist.
- Scientist:                
       - 2 Before
       - 2 After
       - Science! That's what the vaults all about.
- Security Officer:     
       - 3 Before
       - 1 After
       - The vault has no solid laws, and if someone's being a shitter, anyone can deal with it.
- Engineer:               
       - 2 Before
       - 0 After
       - The engineering "department" has full access on its doors, and no real job.
- Dweller:            
       - **_8 Before_**
       - 2 After
       - They serve, no purpose. Just an overflow/miner slot.

And with that, vault goes from 19 jobs, to 9. While that's still a fair few, most SSD, Die, Or more likely, don't even join vault.

## Changelog
:cl:
add: Added vault slots back.
remove: Some vault slots (see this PR)
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

DeveloperLoser committed:
Reverts vault slot removal, keeps other changes.

DeveloperLoser committed:
Re-Adjusts Slots

ghost commented:
The goal was, IIRC, that the vault's closed off for now while it's moved elsewhere. or integrated into vault city.

thedarkcrusader commented:
what about those factions who raid the vault just to  raid the vault that was the only other  reason why they had 3 sec slots. also everyone does not have access to  weapons in order to defend the vault  id be comfortable if they  had at least 2 slots instead of just 1. if your removing HOS remove there office on the map to

DeveloperLoser commented:
> The goal was, IIRC, that the vault's closed off for now while it's moved elsewhere. or integrated into vault city.

The [PR that removed](https://github.com/nokele/desertrose/pull/66) it just listed its overabundance of slots as an issue, in terms of vault city ive heard little on that

> what about those factions who raid the vault just to raid the vault that was the only other reason why they had 3 sec slots. also everyone does not have access to weapons in order to defend the vault id be comfortable if they had at least 2 slots instead of just 1. if your removing HOS remove there office on the map to

This is do-able, but usually the two armory's are just bolted open, or if an attack is underway vault bribes someone for help, or closes the door to buy time.

Pebblesthe2nd commented:
Poorly implemented factions in droves has plagued BD and DR.  Only reason to have a vault is if it is a well done faction that is 
integrated into the gameplay loop. Make a new map based around a actually well made vault that feels like Fallout and not a Nanotrasen base with a few name changes and it will actually add something. 

Only use the vault has as implemented as some sort of dungeon, maybe outbreak of FEV, do something interesting with it. Let this old mangy dog rest in its grave.

DeveloperLoser commented:
> Poorly implemented factions in droves has plagued BD and DR. Only reason to have a vault is if it is a well done faction that is
> integrated into the gameplay loop. Make a new map based around a actually well made vault that feels like Fallout and not a Nanotrasen base with a few name changes and it will actually add something.
> 
> Only use the vault has as implemented as some sort of dungeon, maybe outbreak of FEV, do something interesting with it. Let this old mangy dog rest in its grave.

A map edit for the vault, as well as content addition can be done. But for now this just adds back some slots and tells them not to genocide the surface (Once I actually get around to changing that.) Also pointed out in the discord, is that with no people, vault is a free loot piñata, or an OP player base (can repair the door with 1 weld) so throwing roles back in will fix that.

DeveloperLoser committed:
Modifies Forbid Objective

"Attracting unwanted hostile attention."
Meant to be taken as "Don't fucking genocide people."

DeveloperLoser commented:
Full Forbid objective now reads:
"The Vault forbids: Disobeying the Overseer. Deserting the Vault unless it is rendered unhospitable. Killing fellow Vault Dwellers. **Attracting unwanted hostile attention.**"

As opposed to:
"The Vault forbids: Disobeying the Overseer. Deserting the Vault unless it is rendered unhospitable. Killing fellow Vault Dwellers. Betraying the Vault and its people."

This is meant to be taken as "Don't fucking genocide people", as releasing a deadly virus, or max capping a base would most certainly draw attention and make factions consider eradication better than letting you just go _boop_ against their walls with a TTV

nokele commented:
It's not getting added back unless it is integrated into Oasis or redesigned into a trade vault.

DeveloperLoser commented:
> It's not getting added back unless it is integrated into Oasis or redesigned into a trade vault.

Alright, ill start work on remapping it then

DeveloperLoser committed:
First vault map changes

(Forgot map merge :sob:)
Entrance is now a basic trading area.
Moved the heavy armory.
Started work on damage/decay
